pkgs/tools/misc/gaphor/default.nix

```
@@ -1,52 +1,77 @@
 { lib
 , buildPythonApplication
-, fetchPypi
+, fetchFromGitHub
 , copyDesktopItems
 , gobject-introspection
 , poetry-core
 , wrapGAppsHook
-, gtksourceview4
+, gtksourceview5
+, libadwaita
 , pango
+, babel
+, better-exceptions
+, defusedxml
 , gaphas
 , generic
 , jedi
+, pillow
 , pycairo
+, pydot
+, pygit2
 , pygobject3
 , tinycss2
-, gtk3
+, gtk4
 , librsvg
 , makeDesktopItem
 , python
+, hypothesis
+, ipython
+, pytest-archon
+, sphinx
+, xdoctest
+, pytestCheckHook
 }:
 
 buildPythonApplication rec {
   pname = "gaphor";
-  version = "2.8.2";
+  version = "2.23.2";
+  pyproject = true;
 
-  format = "pyproject";
-
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-+qqsSLjdY2I19fxdfkOEQ9DhTTHccUDll4O5yqtLiz0=";
+  src = fetchFromGitHub {
+    owner = "gaphor";
+    repo = "gaphor";
+    rev = version;
+    hash = "sha256-ZSKhDKEgexQe4XLQrZUIZNCdOR55FFuF6XfJhktobXw=";
   };
 
   nativeBuildInputs = [
     copyDesktopItems
+    # Needed to load namespaces
     gobject-introspection
     poetry-core
     wrapGAppsHook
   ];
 
   buildInputs = [
-    gtksourceview4
+    # Adds Gtk4 namespace
+    gtksourceview5
+    # Adds Adw namespace
+    libadwaita
+    # Adds Pango namespace
     pango
```

```suggestion
    gtksourceview5 # Adds Gtk4 gobject namespace
    libadwaita # Adds Adw gobject namespace
    pango # Adds Pango gobject namespace
```
we can write that a bit compacter and we should mention what namespace